core/trino-main/src/main/java/io/trino/security/AccessControl.java

```diff
@@ -525,12 +526,20 @@ void checkCanRevokeRoles(SecurityContext context,
      */
     void checkCanExecuteFunction(SecurityContext context, String functionName);
 
+    /**
+     * @throws AccessDeniedException if not allowed
+     * @deprecated Use {@link AccessControl#checkCanExecuteFunction(SecurityContext, QualifiedObjectName, FunctionKind)} instead.
+     * Check if identity is allowed to execute function
+     */
+    @Deprecated
+    void checkCanExecuteFunction(SecurityContext context, QualifiedObjectName functionName);
```

Comment:
This method was not used so far. Please simply remove it. The best would be to remove the other method too (the one with function name as `String`)

Reply:
@dain the old method was introduced a short while ago. It was only called for table function invocation analysis, but we don't have any table functions yet. I removed it in a separate commit. If you're OK with removing, I'll squash.

Reply:
@kokosing @kasiafi Do I understand correctly that the newly introduced method
```
    void checkCanExecuteFunction(SecurityContext context, FunctionKind functionKind, QualifiedObjectName functionName);
```
should eventually replace the old one
```
    void checkCanExecuteFunction(SecurityContext context, String functionName);
```
, so that all old functionality for AccessControls remains the same for all FunctionKinds which are not `FunctionKind.TABLE`?